test/nodes/test_retriever.py

```diff
@@ -80,6 +80,18 @@ def test_retrieval(retriever_with_docs: BaseRetriever, document_store_with_docs:
         )
         assert len(result) == 0
 
+    # test FilterRetriever with empty query using the run() method
+    if isinstance(retriever_with_docs, FilterRetriever):
+        result = retriever_with_docs.run(root_node="Query", query="", filters={"name": ["filename2"]})
+        assert len(result) == 2
+        result = result[0]
+        assert "documents" in result
+        result = result["documents"]
+        assert len(result) == 1
+        assert type(result[0]) == Document
+        assert result[0].content == "My name is Paul and I live in New York"
+        assert result[0].meta["name"] == "filename2"
+
```

Comment:
Nitpick: this would be even better if it was an isolated test, instead of being part of `test_retriever`. Let me know if extracting it into a separate test gives you any trouble, I can help out

Reply:
Thanks for the suggestions!
For the `MockRetriever`, should I create a dedicated Mock for the FilterRetriever? Or why is it really needed? I understand that, for instance, DPRs need Mocks because embedding takes time

Reply:
Yes that would require you to create a new mock class. 

On one hand it would make checking the output simpler (no need to check for all those details, you can just `assert result[0] == <the document object you expect>`) and would run regardless of the docstore used (right now FilterRetriever works only with Elasticsearch). On the other hand, this test is so simple that a full mock might be a bit overkill :smile: 

Your call! As long as you put the check into a separate tests it can keep using FilterRetriever, no problem.

Reply:
What do you think about the latest changes? I added a MockFilterRetriever, isolated the test and adapt it to use the mock, does that make sense?

Reply:
I think this code block can be removed now that you added the new test just below. They test the same thing, isn't it?